R/CsvHelper.R

```diff
@@ -0,0 +1,126 @@
+# Copyright 2022 Observational Health Data Sciences and Informatics
+#
+# This file is part of CohortGenerator
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+#' Used to read a csv file
+#'
+#' @description
+#' This function is used to centralize the function for reading
+#' CSV files across the HADES ecosystem.
+#' 
+#' @param file  The csv file to read.
+#'
+#' @return
+#' A tibble with the CSV contents
+#' 
+#' @export
+readCsv <- function(file) {
+  readr::read_csv(file = file, 
+                  col_types = readr::cols(), 
+                  lazy = FALSE)
+}
+
+#' Used to write a csv file
+#'
+#' @description
+#' This function is used to centralize the function for writing
+#' CSV files across the HADES ecosystem. This function may also raise warnings
+#' if the data is stored in a format that will not work with the HADES standard
+#' for uploading to a results database.
+#' 
+#' @param x  A data frame or tibble to write to disk.
+#' 
+#' @param file  The csv file to write.
+#' 
+#' @param warnOnUploadRuleViolations When TRUE, this function will provide
+#' warning messages that may indicate if the data is stored in a format in the
+#' CSV that may cause problems when uploading to a database.
+#' 
+#' @return 
+#'  Returns the input x invisibly.
+#'
+#' @export
+writeCsv <- function(x, file, warnOnUploadRuleViolations = TRUE) {
```

Comment:
I'm wondering if I should change the default of `warnOnUploadRuleViolations` to FALSE so that we only raise warnings when the user wants to know if they may be violating rules?